src/builder-manifest.c

```diff
@@ -3754,41 +3754,25 @@ builder_manifest_show_deps (BuilderManifest *self,
 }
 
 static gboolean
-builder_manifest_install_dep (BuilderManifest *self,
-                              BuilderContext  *context,
-                              const char *remote,
-                              gboolean opt_user,
-                              const char *opt_installation,
-                              const char *runtime,
-                              const char *version,
-                              gboolean opt_yes,
-                              GError         **error)
+builder_manifest_install_single_dep (const char *ref,
+				     const char *remote,
+				     gboolean opt_user,
+				     const char *opt_installation,
+				     GError **error)
 {
-  g_autofree char *ref = NULL;
-  g_autofree char *commit = NULL;
   g_autoptr(GPtrArray) args = NULL;
-  gboolean installed;
-
-  if (version == NULL)
-    version = builder_manifest_get_runtime_version (self);
-
-  ref = flatpak_build_untyped_ref (runtime,
-                                   version,
-                                   builder_context_get_arch (context));
-
-  commit = flatpak_info (opt_user, opt_installation, "--show-commit", ref, NULL);
-  installed = (commit != NULL);
-
   args = g_ptr_array_new_with_free_func (g_free);
   g_ptr_array_add (args, g_strdup ("flatpak"));
   add_installation_args (args, opt_user, opt_installation);
-  if (installed)
+  if (remote == NULL)
```

Comment:
This function is really small now, and it would probably be clearer if it was split into two, one called update_single_dep(), not taking a remote arg, plus this which would always take a remote arg.

Having remote == NULL mean update is kind of unexpected.

Reply:
Hmm, not quite clear what you meant by the function being small. Do you think it is too small? My C "how small is too small" sense is still in development.

Reply:
I mean, its small enough that just duplicating it is fine instead of sharing some stuff and then have a weird if in the middle.

Reply:
Oh, i see your confusion. My wording "split into two" is not great. I mean, create two different functions based on that code, one with the code for the update codepath, and one with the code for the install codepath.

Reply:
Having separate functions does make the code nicer to write as well. Split now. As not-regular-C-coder wasn't sure if tiny functions was a discouraged thing or not.